config/bidderinfo.go

```diff
@@ -177,94 +166,345 @@ type SyncerEndpoint struct {
+	}
+	// Validate the resolved endpoint
+	//
+	// Validating using both IsURL and IsRequestURL because IsURL allows relative paths
+	// whereas IsRequestURL requires absolute path but fails to check other valid URL
+	// format constraints.
+	//
+	// For example: IsURL will allow "abcd.com" but IsRequestURL won't
+	// IsRequestURL will allow "http://http://abcd.com" but IsURL won't
+	if !validator.IsURL(resolvedEndpoint) || !validator.IsRequestURL(resolvedEndpoint) {
+		errs = append(errs, fmt.Errorf("The endpoint: %s for %s is not a valid URL", resolvedEndpoint, bidderName))
 	}
+	return errs
+}
 
-	if s.UserMacro != "" {
-		copy.UserMacro = s.UserMacro
+func validateInfo(info BidderInfo, bidderName string) error {
+	if err := validateMaintainer(info.Maintainer, bidderName); err != nil {
+		return err
+	}
+	if err := validateCapabilities(info.Capabilities, bidderName); err != nil {
+		return err
 	}
 
-	return &copy
+	return nil
 }
 
-// LoadBidderInfoFromDisk parses all static/bidder-info/{bidder}.yaml files from the file system.
-func LoadBidderInfoFromDisk(path string, adapterConfigs map[string]Adapter, bidders []string) (BidderInfos, error) {
-	reader := infoReaderFromDisk{path}
-	return loadBidderInfo(reader, adapterConfigs, bidders)
+func validateMaintainer(info *MaintainerInfo, bidderName string) error {
+	if info == nil || info.Email == "" {
+		return fmt.Errorf("missing required field: maintainer.email for adapter: %s", bidderName)
+	}
+	return nil
 }
 
-func loadBidderInfo(r infoReader, adapterConfigs map[string]Adapter, bidders []string) (BidderInfos, error) {
-	infos := BidderInfos{}
+func validateCapabilities(info *CapabilitiesInfo, bidderName string) error {
+	if info == nil {
+		return fmt.Errorf("missing required field: capabilities for adapter: %s", bidderName)
+	}
 
-	for _, bidder := range bidders {
-		data, err := r.Read(bidder)
-		if err != nil {
-			return nil, err
+	if info.App == nil && info.Site == nil {
+		return fmt.Errorf("at least one of capabilities.site or capabilities.app must exist for adapter: %s", bidderName)
+	}
+
+	if info.App != nil {
+		if err := validatePlatformInfo(info.App); err != nil {
+			return fmt.Errorf("capabilities.app failed validation: %v for adapter: %s", err, bidderName)
 		}
+	}
 
-		info := BidderInfo{}
-		if err := yaml.Unmarshal(data, &info); err != nil {
-			return nil, fmt.Errorf("error parsing yaml for bidder %s: %v", bidder, err)
+	if info.Site != nil {
+		if err := validatePlatformInfo(info.Site); err != nil {
+			return fmt.Errorf("capabilities.site failed validation: %v, for adapter: %s", err, bidderName)
 		}
+	}
+	return nil
+}
 
-		info.Enabled = isEnabledByConfig(adapterConfigs, bidder)
-		infos[bidder] = info
+func validatePlatformInfo(info *PlatformInfo) error {
+	if len(info.MediaTypes) == 0 {
+		return errors.New("mediaTypes should be an array with at least one string element")
```

Comment:
Nitpick: There is no need to include the wording "should be an array". Of course it's an array. Consider changing to an error along the line of "at least one meda type needs to be specified".

Reply:
Modified